### PR TITLE
Add `LambertW` trait for evaluation of Lambert W function

### LINKED ISSUE #14: Adding support for Lambert W function?
I recently finished [an implementation](https://crates.io/crates/lambert_w) of the [method of Toshio Fukushima](https://www.researchgate.net/publication/346309410_Precise_and_fast_computation_of_Lambert_W_function_by_piecewise_minimax_rational_function_approximation_with_variable_transformation) for evaluating the [Lambert W function](https://en.wikipedia.org/wiki/Lambert_W_function) quickly and accurately.

Would it be interesting to integrate this functionality here?

### PR DESCRIPTION
**Pull request by JSorngard**
This PR adds the `LambertW` trait for evaluation of the Lambert W function, and implements it for `f64` by delegating to the `lambert_w` crate. 

This partially implements #14, as it still lacks `f32` support.

It currently does not support using the standard library instead of `libm`, since that would require disabling the `std` feature when this crate enabled the `no_std` feature.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ no_std = []
 
 [dependencies]
 libm = "0.2"
+lambert_w = { version = "0.5.2", default-features = false, features = ["50bits", "libm"] }
```
Comment (IvanUkhov) [inline Cargo.toml:30]: Please put them in alphabetic order.

File: src/omega.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/// Lambert W function.
```
Comment (IvanUkhov) [inline src/omega.rs:1]: If you want to use `LambertW`, let us use it. I realized that there is also `Error`, which is not a Greek letter. So there is no point in sticking to something that might be misleading. And by analogy, the comment here would be “Lambert W functions.” Meaning there are several.

File: src/omega.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    ///
+    /// ```
+    /// use special::Omega;
+    ///
+    /// assert!((1.0.lambert_w0() - 0.5671432904097838).abs() < 1e-15);
+    /// ```
+    ///
+    /// ## References
+    ///
+    /// - T. Fukushima, Precise and fast computation of Lambert W function
```
Comment (IvanUkhov) [inline src/omega.rs:26]: Please use numbers here for consistency with other reference lists:

https://github.com/stainless-steel/special/blob/main/src/gamma.rs#L37

See also the indentation. Not sure if “To be submitted” is relevant. Anyway, you can just leave the formatting to me. I will make it consistent with the rest of the package.

File: src/omega.rs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    /// by piecewise minimax rational function approximation with variable transformation,
+    /// To be submitted, 2020, <https://doi.org/10.13140/RG.2.2.30264.37128>.  
+    /// - The [`lambert_w`] crate.
```
Comment (IvanUkhov) [inline src/omega.rs:29]: This I would move up into the main description as it is done here:

https://github.com/stainless-steel/special/blob/main/src/gamma.rs#L43

File: src/omega.rs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use special::Omega;
+    ///
+    /// assert!(((-f64::ln(2.0) / 2.0).lambert_wm1() + f64::ln(4.0)).abs() < 1e-15);
+    /// ```
+    ///
+    /// ## References
```
Comment (IvanUkhov) [inline src/omega.rs:50]: The same here.

File: src/omega.rs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    ///
+    /// - T. Fukushima, Precise and fast computation of Lambert W function
+    /// by piecewise minimax rational function approximation with variable transformation,
+    /// To be submitted, 2020, <https://doi.org/10.13140/RG.2.2.30264.37128>.  
+    /// - The [`lambert_w`] crate.
+    fn lambert_wm1(self) -> Self;
+}
+
+impl Omega for f64 {
+    fn lambert_w0(self) -> Self {
```
Comment (IvanUkhov) [inline src/omega.rs:60]: Please `#[inline]` here and below.

File: Cargo.toml (hunk 1 above)
Comment (JSorngard) [reply Cargo.toml:29]: Order fixed

File: src/omega.rs (hunk 2 above)
Comment (JSorngard) [reply src/omega.rs]: I changed the name to LambertW

File: src/omega.rs (hunk 3 above)
Comment (JSorngard) [reply src/omega.rs]: I removed "To be submitted" and moved the reference to the external crate into the main description.

File: src/omega.rs (hunk 6 above)
Comment (JSorngard) [reply src/omega.rs]: inlined!

**Comment by JSorngard** [comment]
I think that should be all the changes!  
Let me know if there's anything else.

**Pull request merged**